File: Sistema_v6/frontend/src/lib/encryption.ts
import CryptoJS from 'crypto-js';

// Clave de encriptación desde variable de entorno o default
const ENCRYPTION_KEY = import.meta.env.VITE_ENCRYPTION_KEY || 'default-encryption-key-change-in-production';

/**
 * Encripta un texto usando AES
 * @param text - Texto a encriptar
 * @returns Texto encriptado en base64
 */
export function encryptText(text: string): string {
  try {
    const encrypted = CryptoJS.AES.encrypt(text, ENCRYPTION_KEY);
    return encrypted.toString();
  } catch (error) {
    console.error('Error al encriptar:', error);
    throw new Error('Error al encriptar el texto');
  }
}

/**
 * Desencripta un texto encriptado con AES
 * @param encryptedText - Texto encriptado en base64
 * @returns Texto desencriptado
 */
export function decryptText(encryptedText: string): string {
  try {
    const decrypted = CryptoJS.AES.decrypt(encryptedText, ENCRYPTION_KEY);
    return decrypted.toString(CryptoJS.enc.Utf8);
  } catch (error) {
    console.error('Error al desencriptar:', error);
    throw new Error('Error al desencriptar el texto');
  }
}

/**
 * Genera un hash SHA256 de un texto
 * @param text - Texto a hashear
 * @returns Hash en hexadecimal
 */
export function hashText(text: string): string {
  try {
    return CryptoJS.SHA256(text).toString();
  } catch (error) {
    console.error('Error al generar hash:', error);
    throw new Error('Error al generar hash');
  }
}

File: Sistema_v6/infrastructure/security/encryption.py
"""Utilidades para encriptación de credenciales.

Este módulo proporciona funciones para encriptar y desencriptar credenciales del PJN
usando Fernet (symmetric encryption).
"""

from __future__ import annotations

from cryptography.fernet import Fernet

from infrastructure.config import get_settings

settings = get_settings()

# Obtener o generar la clave Fernet
if settings.encryption.fernet_key:
    _fernet_key = settings.encryption.fernet_key.encode()
else:
    # Generar una nueva clave si no existe
    # NOTA: Esta clave debe guardarse en producción (variable de entorno)
    _fernet_key = Fernet.generate_key()
    print(f"WARNING: Fernet key generada automáticamente: {_fernet_key.decode()}")
    print("Guarde esta clave en la variable de entorno ENCRYPTION_FERNET_KEY")

cipher_suite = Fernet(_fernet_key)


def encrypt_credentials(text: str) -> str:
    """Encripta un texto usando Fernet.

    Args:
        text: Texto a encriptar (credencial del PJN)

    Returns:
        Texto encriptado (base64)

    Example:
        >>> encrypted = encrypt_credentials("mi_password")
        >>> print(encrypted)
        gAAAAABk...
    """
    if not text:
        return ""
    return cipher_suite.encrypt(text.encode()).decode()


def decrypt_credentials(encrypted_text: str) -> str:
    """Desencripta un texto usando Fernet.

    Args:
        encrypted_text: Texto encriptado (base64)

    Returns:
        Texto desencriptado

    Example:
        >>> encrypted = encrypt_credentials("mi_password")
        >>> decrypted = decrypt_credentials(encrypted)
        >>> print(decrypted)
        mi_password
    """
    if not encrypted_text:
        return ""
    return cipher_suite.decrypt(encrypted_text.encode()).decode()


def generate_fernet_key() -> str:
    """Genera una nueva clave Fernet.

    Returns:
        Clave Fernet en formato base64 string

    Example:
        >>> key = generate_fernet_key()
        >>> print(key)
        abcdef123456...
    """
    return Fernet.generate_key().decode()
